app/controllers/issues_controller.rb

```diff
@@ -15,9 +16,10 @@ class IssuesController < AuthenticatedController
   before_action :set_or_initialize_tags, except: [:destroy]
   before_action :set_auto_save_key, only: [:new, :create, :edit, :update]
 
+  EXTRA_COLUMNS = ['Title', 'Tags', 'Affected', 'Created', 'Created by', 'Updated'].freeze
+
   def index
-    @columns = @issues.map(&:fields).map(&:keys).uniq.flatten |
-      ['Title', 'Tags', 'Affected', 'Created', 'Created by', 'Updated']
+    @columns = cached_collection_column_keys(@unsorted_issues) | EXTRA_COLUMNS
```

Comment:
I get that the first time around you don't have anything, but cache hitting / building every single time you go back to the page seems a bit much?

Should be also do the calculation post #update or #create to try to minimise the need to build on #index?

Reply:
> but cache hitting every single time you go back to the page seems a bit much?

Isn't that the definition and purpose of caching? We look for a cache hit and act with it. That's pretty SOP.

>  building every single time you go back

Only gets built on cache miss. So still winning.

> Should be also do the calculation post #update or #create to try to minimise the need to build on #index?
Answer on pacifica.